synapse/storage/databases/main/account_data.py

```diff
@@ -71,6 +89,7 @@ def __init__(self, database: DatabasePool, db_conn, hs: "HomeServer"):
             # updated over replication. (Multiple writers are not supported for
             # SQLite).
             if hs.get_instance_name() in hs.config.worker.writers.account_data:
+                self._can_write_to_account_data = True
```

Comment:
It looks like the previous code incorrectly set `_can_write_to_account_data` to `True` in sqlite mode, even when the current worker wasn't supposed to write.

Reply:
Yes, I suspect the code's never been tested in sqlite mode on a non-writer instance.
It certainly can't work, since all the methods that do writes expect an id generator.